cupy/core/core.pyx

```diff
@@ -1698,11 +1706,21 @@ cpdef ndarray _take(ndarray a, indices, axis=None, ndarray out=None):
         if out is None:
             return a[indices].copy()
         else:
-            out[:] = a[indices]
+            if out.dtype != a.dtype:
+                raise TypeError('Output dtype mismatch')
+            if out.shape != a.shape[1:]:
+                raise ValueError('Output shape mismatch')
+            out[()] = a[indices]
             return out
     elif not isinstance(indices, ndarray):
         indices = array(indices, dtype=int)
 
+    if not ((-index_range <= indices) & (indices < index_range)).all():
```

Comment:
This check is to decrease the performance.
Please comment out this code.
We may provide a debug mode.